WTG.Analyzers/Rules/Rules.xml

```diff
@@ -124,6 +124,11 @@
 			<message name="None">The {opcode} opcode does not take an argument.</message>
 			<description>The emit methods generally don't pay attention to the opcode when emitting the operand, so using the wrong overload may result in unexpected behaviour that is difficult to debug.</description>
 		</rule>
+		<rule id="6" name="ForbidCompiledInStaticRegexMethods" severity="Warning">
+			<title>Do not pass the Compile option into static methods on Regex.</title>
+			<message>Do not use the Compile option when calling static methods on Regex.</message>
+			<description>The static methods on Regex will discard the regex object, largely negating the benifits of the Compiled option and leaving you with just the increased costs.</description>
```

Comment:
It's not entirely discarded. .NET Framework and Core both cache recent regexes internally... though if you're compiling a regex and relying on the cache, you should probably just cache it yourself.

Reply:
They discard the Regex object, but cache the innards. But even then, the innards still get discarded if you create enough Regex objects with different patterns before recreating a Regex object with the same pattern. :)

Reply:
Yeah, the default seems to be the most recent 15 regexes.